Server/Scripts/configure.ps1

```diff
@@ -3,167 +3,126 @@ $env:OCTOPUS_INSTANCENAME=$OctopusInstanceName
 
 . ./common.ps1
-
-  #remove any pre-release suffixes - for our purposes, the major.minor.patch is sufficient
-  
+ 
   $sqlDbConnectionString = $env:sqlDbConnectionString
   $masterKey = $env:masterKey
-  $masterKeySupplied = ($masterKey -ne $null) -and ($masterKey -ne "")
   $octopusAdminUsername = $env:OctopusAdminUsername
   $octopusAdminPassword = $env:OctopusAdminPassword
-  $credentialsSupplied=($octopusAdminPassword -ne $null -or $octopusAdminUserName -ne $null)
-  $configFile = "c:\Octopus\OctopusServer.config"
-  $ServerNodeName = $env:ServerNodeName
+  $octopusAdminEmail = $env:OctopusAdminEmail
   
-function Validate-Variables() {
-    # If either a username or password has been set, the other is set to a default. We also set
-    # default creds if no master key is supplied.
-    if ($credentialsSupplied -or !$masterKeySupplied)
-    {
-      if ($octopusAdminUsername -eq $null)
-      {
-        $script:octopusAdminUsername = "admin"
-      }
-      if ($octopusAdminPassword -eq $null)
-      {
-        $script:octopusAdminPassword = "Passw0rd123"
-      }
-      Write-Log " - local admin user '$octopusAdminUsername'"
-      Write-Log " - local admin password '##########'"
-    }
 
-    if($sqlDbConnectionString -eq $null){
-        Write-Error "Environment variable sqlDbConnectionString required"
-        Exit 2
-    }
-    $maskedConnectionString = "$sqlDbConnectionString;" -replace "password=.*?;", "password=###########;"
-    Write-Log " - using database '$maskedConnectionString'"
-  }
+function Create-Instance() {
+  Write-Log "Creating Octopus Deploy instance ..."
+  Execute-Command $Exe @('create-instance', '--instance', $OctopusInstanceName, '--config', 'C:\Octopus\OctopusServer.config')
+}
 
-  function Move-Logs() {
-    # Move the log files back in case it was mounted. The files were moved during the install phase
-    # Mounting windows containers requires the volumes to be empty
-    # https://github.com/docker/for-win/issues/644
-    Write-Log "Moving Octopus configuration back from temporary location"
-    if(!(Test-Path C:\Octopus\Logs)) {
-      mkdir C:\Octopus\Logs | Out-Null
-    }
-    mv C:\Octopus\LogsTemp\* C:\Octopus\Logs
-    Remove-Item C:\Octopus\LogsTemp -Recurse -Force
-    Write-Log "moved"
+function Configure-Database() {
+  if($null -eq $sqlDbConnectionString){
+    Write-Error "Environment variable sqlDbConnectionString required"
+    Exit 2
   }
+  $maskedConnectionString = "$sqlDbConnectionString;" -replace "password=.*?;", "password=###########;"
+  Write-Log " - using database '$maskedConnectionString'"
+  Write-Log "Creating Octopus Deploy database ..."
+  $args = @('database','--instance', $OctopusInstanceName, '--connectionString', $sqlDbConnectionString, '--create')
+  if ($null -ne $masterKey) {
+    $args += @('--masterkey', $masterKey)
+  }
+  Execute-Command $Exe $args @($masterKey, $sqlDbConnectionString)
+  Export-MasterKey
```

Comment:
Consider doing this in the main block, rather than from within `Configure-Database`